### ISSUE TITLE
Hyphen-minus ("-") in file names produces invalid C++ identifiers

### ISSUE THREAD

**Issue by replomancer** [RosettaCommons/binder#103]
Hello!

I'm trying to generate bindings for a C++ library where some folder and file names contain the "-" character. Based on the file names Binder produces invalid C++ identifiers.

Example:
When I try to bind code from a `demo-project` folder and my header file is called `demo-1.h`, generation of bindings succeeds without errors, but building attempt results in:

```
sourcedemoproject.cpp:11:15: error: expected initializer before ‘-’ token
 void bind_demo-project_demo-1(std::function< pybind11::module &(std::string const &namespace_) > &M);
               ^
sourcedemoproject.cpp: In function ‘void pybind11_init_demoproject(pybind11::module&)’:
sourcedemoproject.cpp:33:2: error: ‘bind_demo’ was not declared in this scope
  bind_demo-project_demo-1(M);
  ^~~~~~~~~
sourcedemoproject.cpp:33:12: error: ‘project_demo’ was not declared in this scope
  bind_demo-project_demo-1(M);
            ^~~~~~~~~~~~
sourcedemoproject.cpp:33:28: error: expression cannot be used as a function
  bind_demo-project_demo-1(M);
```

I'm guessing this is a known issue, but I couldn't find it here.
Is there a workaround other than renaming/editing files?

**Comment by lyskov** [comment]
Thank you for letting me know @replomancer! I will looks this up.

**Comment by lyskov** [comment]
fixed in 170978f
